lib/report_portal/cucumber/report.rb

```diff
@@ -46,19 +47,45 @@ def initialize
       end
 
       def start_launch(desired_time = ReportPortal.now, cmd_args = ARGV)
+        # Not sure what is the use case if launch id is missing. But it does not make much of practical usage
+        #
+        # Expected behavior that make sense:
+        #  1. If launch_id present attach to existing (simple use case)
+        #  2. If launch_id not present check if exist rp_launch_id.tmp
+        #  3. [ADDED] If launch_id is not present check if lock exist with launch_uuid
         if attach_to_launch?
           ReportPortal.launch_id =
               if ReportPortal::Settings.instance.launch_id
                 ReportPortal::Settings.instance.launch_id
               else
-                file_path = ReportPortal::Settings.instance.file_with_launch_id || (Pathname(Dir.tmpdir) + 'rp_launch_id.tmp')
-                File.read(file_path)
+                file_path = lock_file
+                File.file?(file_path) ? read_lock_file(file_path) : new_launch(desired_time, cmd_args, file_path)
```

Comment:
Sorry, but I still don't see why it makes sense to create a new launch when `attach_to_launch` is used.
The idea of attach_to_launch was different.
The idea was to create a launch e.g. via rake task before starting execution of features.
And after all tests finish, finish the launch explicitly via e.g. a rake task.
I see that you try to do something different.

Could you explain the use case? I'm not sure it makes sense to use `attach_to_launch` for behavior you try to achieve.

Reply:
ok, when you simply run `parallel_cucumber -o '-p smoke'` why does user need to create launch separately?

Reply:
The user doesn't have to create a launch separately.
If the users wants launch to be created, they should use the gem without `attach_to_launch` formatter mode.

Reply:
do you understand how parallel_cucumber start cucumber? how do you see it possible?